add option for deserializing binary into Buffer object

msmiley opened the pull request:
**promoteBuffers** {Boolean, default:false}

christkv commented:
would need to be also applied to bson-ext module in c++ to be pulled in and include tests.

tellnes commented on lib/bson/bson.js:
```diff
@@ -1282,8 +1284,13 @@ BSON.deserialize = function(buffer, options, isArray) {
-          object[name] = new Binary(buffer.slice(index, index + binarySize), subType);
+          if(nativeBuffers) {
+            // Slice the data and create a native Buffer object
+            object[name] = new Buffer(buffer.slice(index, index + binarySize));
```
`buffer.slice()` returns a buffer, why not just use that? Now you are doing a memory copy. https://github.com/nodejs/io.js/blob/master/lib/buffer.js#L96-L100

msmiley replied:
you are absolutely right, thanks for catching that!

msmiley commented:
sorry for taking so long on this, rebased and changed the name of the variable to promoteBuffers to be more in line with the existing promoteLongs option. Also, nativeBuffers may make some users think C++.

Added a couple tests, let me know if I need to add a test somewhere else.

Will make a PR on bson-ext.
